mkosi/__init__.py

```diff
@@ -883,25 +885,49 @@ def prepare_xbootldr(args: CommandLineArguments, loopdev: Optional[str], cached:
         run(["mkfs.fat", "-nXBOOTLDR", "-F32", partition(loopdev, args.xbootldr_partno)])
 
 
-def mkfs_ext4(label: str, mount: str, dev: str) -> None:
-    run(["mkfs.ext4", "-I", "256", "-L", label, "-M", mount, dev])
+def mkfs_ext4(label: str, mount: str, dev: str, args: Optional[List[str]]) -> None:
```

Comment:
> Maybe do `args: List[str] = []` instead. 

This is something that should generally be avoided, because Python initialises default arguments only once, i.e. if the list were mutated inside the function, these would persist function invocations. Only primitive types make good defaults. The idiomatic way to circumvent this is assigning `None` as the default and pushing initialisation into the function body.

Reply:
Reworked the `mkfs_<filesystem>()` methods to `mkfs_<filesystem>_cmd()` as suggested, please have a look.